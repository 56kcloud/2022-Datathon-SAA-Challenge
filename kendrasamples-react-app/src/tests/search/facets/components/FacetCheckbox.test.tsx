import { shallow } from "enzyme";
import _ from "lodash";
import React from "react";
import { FacetCheckbox } from "../../../../search/facets/components/FacetCheckbox";

const facetConfiguration = {
  showCount: true,
};

const mockDocumentAttributeValue = {
  StringValue: "mockStringValue1",
};

describe("A <FacetCheckbox />", () => {
  it("renders correctly", () => {
    const component = shallow(
      <FacetCheckbox
        value={mockDocumentAttributeValue}
        valueLabel={mockDocumentAttributeValue.StringValue}
        count={5}
        selected={false}
        disableAdd={false}
        facetConfiguration={facetConfiguration}
        onSelectionChange={_.noop}
        attributeName={"mockAttributeName"}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("renders without count correctly", () => {
    const component = shallow(
      <FacetCheckbox
        value={mockDocumentAttributeValue}
        valueLabel={mockDocumentAttributeValue.StringValue}
        selected={false}
        disableAdd={false}
        facetConfiguration={{
          showCount: false,
        }}
        onSelectionChange={_.noop}
        attributeName={"mockAttributeName"}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("renders disableAdd selected correctly", () => {
    const component = shallow(
      <FacetCheckbox
        value={mockDocumentAttributeValue}
        valueLabel={mockDocumentAttributeValue.StringValue}
        count={5}
        selected={true}
        disableAdd={true}
        facetConfiguration={facetConfiguration}
        onSelectionChange={_.noop}
        attributeName={"mockAttributeName"}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("renders disable not selected correctly", () => {
    const component = shallow(
      <FacetCheckbox
        value={mockDocumentAttributeValue}
        valueLabel={mockDocumentAttributeValue.StringValue}
        count={5}
        selected={false}
        disableAdd={true}
        facetConfiguration={facetConfiguration}
        onSelectionChange={_.noop}
        attributeName={"mockAttributeName"}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
