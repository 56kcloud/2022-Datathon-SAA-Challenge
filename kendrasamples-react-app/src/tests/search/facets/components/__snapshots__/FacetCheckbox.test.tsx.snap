// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A <FacetCheckbox /> renders correctly 1`] = `
<div
  className="facet-checkbox"
>
  <FormCheck
    checked={false}
    disabled={false}
    inline={false}
    isInvalid={false}
    isValid={false}
    label={
      <span
        onClick={[Function]}
      >
        <span>
          mockStringValue1
           (
          5
          )
        </span>
      </span>
    }
    onChange={[Function]}
    title=""
    type="checkbox"
  />
</div>
`;

exports[`A <FacetCheckbox /> renders disable not selected correctly 1`] = `
<div
  className="facet-checkbox"
>
  <FormCheck
    checked={false}
    disabled={true}
    inline={false}
    isInvalid={false}
    isValid={false}
    label={
      <span
        onClick={[Function]}
      >
        <span>
          mockStringValue1
           (
          5
          )
        </span>
      </span>
    }
    onChange={[Function]}
    title=""
    type="checkbox"
  />
</div>
`;

exports[`A <FacetCheckbox /> renders disableAdd selected correctly 1`] = `
<div
  className="facet-checkbox"
>
  <FormCheck
    checked={true}
    disabled={false}
    inline={false}
    isInvalid={false}
    isValid={false}
    label={
      <span
        onClick={[Function]}
      >
        <span>
          mockStringValue1
           (
          5
          )
        </span>
      </span>
    }
    onChange={[Function]}
    title=""
    type="checkbox"
  />
</div>
`;

exports[`A <FacetCheckbox /> renders without count correctly 1`] = `
<div
  className="facet-checkbox"
>
  <FormCheck
    checked={false}
    disabled={false}
    inline={false}
    isInvalid={false}
    isValid={false}
    label={
      <span
        onClick={[Function]}
      >
        mockStringValue1
      </span>
    }
    onChange={[Function]}
    title=""
    type="checkbox"
  />
</div>
`;
